extensions/arc/src/ui/dialogs/connectControllerDialog.ts

```diff
@@ -108,14 +117,20 @@ export class ConnectToControllerDialog extends InitializingComponent {
 		}
 		const controllerInfo: ControllerInfo = {
 			url: url,
+			name: this.nameInputBox.value ?? '',
 			username: this.usernameInputBox.value,
 			rememberPassword: this.rememberPwCheckBox.checked ?? false,
 			resources: []
 		};
 		const controllerModel = new ControllerModel(this._treeDataProvider, controllerInfo, this.passwordInputBox.value);
 		try {
-			// Validate that we can connect to the controller
+			// Validate that we can connect to the controller, this also populates the controllerRegistration from the connection response.
 			await controllerModel.refresh(false);
+			if (!controllerModel.controllerRegistration?.instanceName) {
+				throw new Error(`Unknown Error. controllerRegistration.instanceName should be defined after controllerModel is refreshed`);
```

Comment:
Don't throw an error here - just give it some reasonable default. I wouldn't expect this to ever actually happen but if it does happen the user can't do anything about it so no reason to block them from continuing since this is just cosmetic.

Reply:
Good suggestion. Thanks. Changed